### PR TITLE
persist url for SAML redirection

### PR DESCRIPTION
**Pull request by mike-marcacci**
Currently, logging in with SAML loses parameters stored in the the URL, including those necessary for OAuth from a client. This persists the current URL in a cookie before performing the redirect to the IDP.

Additionally, this changes the cookie names such that they are namespaced by the strategy identifier. This should help avoid conflicts and provide more predictable naming for cookies across strategies/plugins.

### CODE REVIEW

File: packages/strategy-saml/src/interface/SamlAuthority.tsx [hunk 1]
```diff
@@ -45,8 +45,9 @@ export function SamlAuthority({
           id: cookieMap.samlFinishedAuthorizationId,
```
Comment (ebrown32) [inline packages/strategy-saml/src/interface/SamlAuthority.tsx:45]:
```suggestion
          id: cookieMap["strategy.saml.authorization_id"],
```

File: packages/strategy-saml/src/interface/SamlAuthority.tsx [hunk 2]
```diff
@@ -45,8 +45,9 @@ export function SamlAuthority({
...
           secret: cookieMap.samlFinishedAuthorizationSecret,
```
Comment (ebrown32) [inline packages/strategy-saml/src/interface/SamlAuthority.tsx:46]:
```suggestion
          secret: cookieMap["strategy.saml.authorization_secret"],
```

File: packages/strategy-saml/src/interface/SamlAuthority.tsx (hunk 1 above)
Comment (ebrown32) [inline packages/strategy-saml/src/interface/SamlAuthority.tsx:45]: It also needs similar changes on line 41 and 42.

**Review by ebrown32 — CHANGES_REQUESTED** [review]

**Comment by mike-marcacci** [comment]
🤦 wow, great catch... not sure how my find/replace missed that!

**Review by ebrown32 — APPROVED** [review]

**Pull request merged**